### ISSUE TITLE
Rouge score Text Summarization

### ISSUE THREAD

**Issue by tahmedge** [microsoft/MASS#42]
Hi, how can I calculate the Rouge score of text summarization after fine tuning? Currently, it is showing acc, bleu other scores after fine tuning.

**Comment by david-liao** [comment]
I have the same question. It seems there is no code for computing rouge score in the source code provided. I use rouge package 
https://github.com/pltrdy/rouge to calculate the rouge score.

**Comment by hivestrung** [comment]
> I have the same question. It seems there is no code for computing rouge score in the source code provided. I use rouge package
> https://github.com/pltrdy/rouge to calculate the rouge score.

Hi David, could you perhaps explain how to calculate the Rouge Score?

Also, the reason I'm asking this is that I want to modify the loss function during training. I'm working on a text summarisation project and I want to be able to control the length of a summary. So I was considering getting the model to, at each training step, output a summary for a given input, and penalise it using a MSE for the length difference.

**Comment by bcmi220** [comment]
> I have the same question. It seems there is no code for computing rouge score in the source code provided. I use rouge package
> https://github.com/pltrdy/rouge to calculate the rouge score.

I also use this to calculate the rouge score, but it outputs a total different score on the pretrained model.

files2rouge ./result/test.ar-ti.ti.desub.pred ./data/sumdata/Giga/task1_ref0.txt

**Comment by StillKeepTry** [comment]
I use files2rouge to calc rouge, and a copy of cnndm fine-tuned model can be downloaded from [here](https://modelrelease.blob.core.windows.net/mass/cnndm_evaluation.tar.gz)

**Comment by cmeister747** [comment]
Hi @StillKeepTry, thank you for all of your responses! I'm working against a deadline and trying to reproduce some results. Once you generate summarizations from a model, how are you getting them into a proper format for files2rouge? The output from fairseq has a lot of extra text and the reference summarizations that are output still have BPE encoding so it doesn't seem very straightforward to get the data into a "reference" and "hypothesis" format that files2rouge needs

**Comment by RebolloM** [comment]
> Hi @StillKeepTry, thank you for all of your responses! I'm working against a deadline and trying to reproduce some results. Once you generate summarizations from a model, how are you getting them into a proper format for files2rouge? The output from fairseq has a lot of extra text and the reference summarizations that are output still have BPE encoding so it doesn't seem very straightforward to get the data into a "reference" and "hypothesis" format that files2rouge needs

**Comment by RebolloM** [comment]
I have the same question
